src/idl_gen_general.cpp

```diff
@@ -1212,7 +1212,8 @@ void GenStruct(StructDef &struct_def, std::string *code_ptr) {
       code += "); }\n";
     }
   }
-  if (struct_def.has_key) {
+  if (struct_def.has_key && (lang_.language == IDLOptions::kJava ||
+    lang_.language == IDLOptions::kCSharp)) {
```

Comment:
When the Go version will be released, absence of this check will cause errors. Remove it?